### ISSUE TITLE
Delay JS - WP Smart Preloader breaks the GTMetrix/PageSpeed Insights tests

### ISSUE THREAD

**Issue by vmanthos** [wp-media/wp-rocket#4708]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅`3.7.10`
- Used the search feature to ensure that the bug hasn’t been reported before ✅

**Describe the bug**

When:
1. The [WP Smart Preloader](https://wordpress.org/plugins/wp-smart-preloader/) is installed and a preloader is applied. - 10,000+ active installations
2. WP Rocket's Delay JavaScript Execution feature is active

both GTMetrix:
![](https://i.vgy.me/Dm81LZ.jpg)

and PageSpeed Insights:
![](https://i.vgy.me/6yth0B.jpg)

throw errors because no content has been painted.

The issue is resolved by excluding the following files from the Delay JavaScript Execution feature:
```
/jquery-?[0-9.](.*)(.min|.slim|.slim.min)?.js
/wp-smart-preloader/assets/js/wsp-main-script.js
```

Given that `jQuery` has to be excluded as well, we should create a compatibility file for this.

**To Reproduce**

1.  Install [WP Smart Preloader](https://wordpress.org/plugins/wp-smart-preloader/)
2. Activate **Delay JavaScript Execution**
3. Run GTMetrix/PageSpeed Insights test
4. See error.

**Expected behavior**

No errors should be displayed in tests when Delay JavaScript Execution is used with WP Smart Preloader.

**Additional context**

**Related ticket:** https://secure.helpscout.net/conversation/1773758583/323066/

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by piotrbak** [comment]
Will be fixed with the 3.13 backend

**Comment by piotrbak** [comment]
Handled in 3.13 using dynamic lists
